### PR TITLE
Fix integer environment parsing

### LINKED ISSUE #412: get_env<int> returns uninitialized data for malformed values
## Summary

`get_env<int>` ignores the return value of `std::sscanf` and returns an uninitialized local when an environment variable is set to a non-numeric value:

```cpp
int value;
std::sscanf(c_str, "%d", &value);
return value;
```

If parsing fails, `value` is never initialized and reading it is undefined behavior. The unset case is safe because it returns the configured default before reaching this branch.

## Impact

The helper is used by more than twenty C++ configuration reads. Most are debug flags, where malformed values can unexpectedly enable logging. `DG_JIT_CPP_STANDARD` is more consequential: a plausible value such as `c++17` fails `%d` parsing and can put an indeterminate standard number into the JIT compiler command.

The same parser bug was independently reported in deepseek-ai/DeepEP#733.

## Expected behavior

Integer environment variables should be parsed deterministically. Malformed, partially parsed, and out-of-range values should fail clearly instead of returning uninitialized or truncated data.

## Proposed fix

Use a full-consumption, range-checked integer parser and cover it with a host-only C++ regression test. The test should require no CUDA toolkit or GPU and should cover valid signs/whitespace/bounds plus empty, boolean, hex-like, trailing-junk, and overflow inputs.

### PR DESCRIPTION
**Pull request by OllieinCanada**
Fixes #412

## Summary

Make integer environment-variable parsing deterministic and reject malformed or
out-of-range values.

## Root cause

`get_env<int>()` ignored the return value from `std::sscanf()` and returned an
uninitialized local variable when an environment value did not begin with a
valid integer. Partial values such as `16abc` were also silently accepted.

## Fix

- parse base-10 integers with `std::from_chars()` and require full consumption
- preserve the existing acceptance of surrounding ASCII whitespace and a
  leading `+`
- reject empty, malformed, partial, and out-of-range values with an error that
  identifies the environment variable
- add a dependency-free host test covering valid boundaries and failure cases

This does not change unset-variable defaults or string environment variables.

## Validation

The host-only regression test passed with GCC 9.4 in C++17 mode using:

```text
g++ -std=c++17 -O2 -Wall -Wextra -Werror -pedantic \
    -fsanitize=undefined -fno-sanitize-recover=all \
    tests/test_env.cpp -I. -o /tmp/deepgemm-test-env
/tmp/deepgemm-test-env
```

The test covers zero, signed values, surrounding whitespace, `INT_MIN`,
`INT_MAX`, empty input, whitespace-only input, malformed booleans, partial
numbers, hex-like input, and positive/negative overflow. It requires no CUDA,
GPU, model, network access, or submodules.

The CUDA/GPU test suite was not run in this environment.

### CODE REVIEW

File: csrc/utils/env.hpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    if (begin == end)
+        return std::nullopt;
+
+    // Unlike scanf and strtol, from_chars does not accept a leading plus.
+    // Preserve that conventional integer spelling explicitly.
+    if (*begin == '+') {
+        ++begin;
+        if (begin == end)
+            return std::nullopt;
+    }
```
[the remark this answers is not in the document]
Comment (OllieinCanada) [reply csrc/utils/env.hpp:33]: Fixed in ea051a0. After removing a leading `+`, the parser now requires the next character to be a decimal digit, so duplicate/mixed signs are rejected before `from_chars`. Added regression cases for `++5`, `+-1`, `+-2147483648`, `--5`, and internal whitespace; the host-only test passes with warnings denied and UBSan.

**Comment by OllieinCanada** [comment]
Closing as superseded by the current main architecture. DeepGEMM removed csrc/utils/system.hpp and now delegates environment parsing to deep_jit::get_env via the pinned DeepJIT submodule (e5bdee2). That implementation uses std::from_chars and requires complete input consumption, so malformed and partial integer values no longer reach an uninitialized result. Rebasing this PR would restore a deleted utility and duplicate the replacement parser. The original #412 failure is therefore addressed by the DeepJIT migration rather than this branch.
